compiler/optimizer/Inliner.cpp

```diff
@@ -3087,6 +3087,11 @@ TR_HandleInjectedBasicBlock::createTemps(bool replaceAllReferences)
             if (tt->getNode()->getOpCode().isBranch() || tt->getNode()->getOpCode().isSwitch())
                tt = tt->getPrevTreeTop();
 
+            // If this treetop is an OSR point, a store cannot be placed between it and the 
+            // transition treetop in postExecutionOSR
```

Comment:
getOSRTransitionTreeTop will consider the mode and act correspondingly, returning the provided treetop in the preExecutionOSR case.